source/plugins/ruby/KubernetesApiClient.rb

```diff
@@ -396,42 +401,67 @@ def getPodUid(podNameSpace, podMetadata)
       return podUid
     end
 
-    def getContainerResourceRequestsAndLimits(metricJSON, metricCategory, metricNameToCollect, metricNametoReturn, metricTime = Time.now.utc.iso8601)
+    def getContainerResourceRequestsAndLimits(pod, metricCategory, metricNameToCollect, metricNametoReturn, metricTime = Time.now.utc.iso8601)
       metricItems = []
       begin
         clusterId = getClusterId
-        metricInfo = metricJSON
-        metricInfo["items"].each do |pod|
-          podNameSpace = pod["metadata"]["namespace"]
-          podUid = getPodUid(podNameSpace, pod["metadata"])
-          if podUid.nil?
-            next
-          end
-
-          # For ARO, skip the pods scheduled on to master or infra nodes to ingest
-          if isAROV3Cluster() && !pod["spec"].nil? && !pod["spec"]["nodeName"].nil? &&
-             (pod["spec"]["nodeName"].downcase.start_with?("infra-") ||
-              pod["spec"]["nodeName"].downcase.start_with?("master-"))
-            next
-          end
+        podNameSpace = pod["metadata"]["namespace"]
+        podUid = getPodUid(podNameSpace, pod["metadata"])
+        if podUid.nil?
+          return metricItems
+        end
 
-          podContainers = []
-          if !pod["spec"]["containers"].nil? && !pod["spec"]["containers"].empty?
-            podContainers = podContainers + pod["spec"]["containers"]
-          end
-          # Adding init containers to the record list as well.
-          if !pod["spec"]["initContainers"].nil? && !pod["spec"]["initContainers"].empty?
-            podContainers = podContainers + pod["spec"]["initContainers"]
-          end
+        nodeName = ""
+        #for unscheduled (non-started) pods nodeName does NOT exist
+        if !pod["spec"]["nodeName"].nil?
+          nodeName = pod["spec"]["nodeName"]
+        end
+        # For ARO, skip the pods scheduled on to master or infra nodes to ingest
+        if isAROv3MasterOrInfraPod(nodeName)
```

Comment:
these are all un-necessary changes, that just triggers testing ARO. I see the same for features like GPU. Its ok to touch, but please test & validate them

Reply:
As we discussed in the email, these are changes as part of emitting smaller streams and perf optimizations. Sure, will ensure no regressions.